src/System.Data.OleDb/src/System/Data/ProviderBase/DbConnectionFactory.cs

```diff
@@ -0,0 +1,591 @@
+            return newConnection;
+        }
+
+        internal DbConnectionInternal CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions) {
+            Debug.Assert(null != pool, "null pool?");
+            DbConnectionPoolGroupProviderInfo poolGroupProviderInfo = pool.PoolGroup.ProviderInfo;
+
+            DbConnectionInternal newConnection = CreateConnection(options, poolKey, poolGroupProviderInfo, pool, owningObject, userOptions);
+            if (null != newConnection) {
+                PerformanceCounters.HardConnectsPerSecond.Increment();
+                newConnection.MakePooledConnection(pool);
+            }
+            // Bid.Trace("<prov.DbConnectionFactory.CreatePooledConnection|RES|CPOOL> %d#, Pooled database connection created.\n", ObjectID);
+            return newConnection;
+        }
+
+        virtual internal DbConnectionPoolGroupProviderInfo CreateConnectionPoolGroupProviderInfo (DbConnectionOptions connectionOptions) {
+            return null;
+        }
+
+        private Timer CreatePruningTimer() {
+            TimerCallback callback = new TimerCallback(PruneConnectionPoolGroups);
+            return new Timer(callback, null, PruningDueTime, PruningPeriod);
+        }
+
+        protected DbConnectionOptions FindConnectionOptions(DbConnectionPoolKey key) {
+            Debug.Assert(key != null, "key cannot be null");
+            if (!ADP.IsEmpty(key.ConnectionString)) {
+                DbConnectionPoolGroup connectionPoolGroup;
+                Dictionary<DbConnectionPoolKey, DbConnectionPoolGroup> connectionPoolGroups = _connectionPoolGroups;
+                if (connectionPoolGroups.TryGetValue(key, out connectionPoolGroup)) {
+                    return connectionPoolGroup.ConnectionOptions;
+                }
+            }
+            return null;
+        }
+
+        // GetCompletedTask must be called from within s_pendingOpenPooled lock
+        static Task<DbConnectionInternal> GetCompletedTask()
+        {
+            if (s_completedTask == null) {
+                TaskCompletionSource<DbConnectionInternal> source = new TaskCompletionSource<DbConnectionInternal>();
+                source.SetResult(null);
+                s_completedTask = source.Task;
+            }
+            return s_completedTask;
+        }
+
+        internal bool TryGetConnection(DbConnection owningConnection, TaskCompletionSource<DbConnectionInternal> retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, out DbConnectionInternal connection) {
+            Debug.Assert(null != owningConnection, "null owningConnection?");
+
+            DbConnectionPoolGroup poolGroup;
+            DbConnectionPool connectionPool;
+            connection = null;
+
+            // SQLBU 431251: 
```

Comment:
Bug numbers -- I think these comments are probably useful to retain , just not the bug #'s